pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -0,0 +1,224 @@
+	// Create Lighthouse-agent
+	lighthouseAgent := newLighthouseAgent(instance)
+
+	// Set ServiceDiscovery instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, lighthouseAgent, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if this lighthouse-agent already exists
+	found := &corev1.Pod{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: lighthouseAgent.Name, Namespace: lighthouseAgent.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new lighthouse-agent", "Pod.Namespace", lighthouseAgent.Namespace, "Pod.Name", lighthouseAgent.Name)
+		err = r.client.Create(context.TODO(), lighthouseAgent)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Lighthouse-agent created successfully - don't requeue
+		return reconcile.Result{}, nil
+	} else if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// lighthouse-agent already exists - don't requeue
+	reqLogger.Info("Skip reconcile: Pod already exists", "Pod.Namespace", found.Namespace, "Pod.Name", found.Name)
+	return reconcile.Result{}, nil
+}
+
+func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.DaemonSet {
+	labels := map[string]string{
+		"app":       "submariner-lighthouse-agent",
+		"component": "lighthouse-agent",
+	}
+
+	matchLabels := map[string]string{
+		"app": "submariner-lighthouse-agent",
+	}
+
+	allowPrivilegeEscalation := true
+	privileged := true
+	readOnlyFileSystem := false
+	runAsNonRoot := false
+	security_context_all_cap_allow_escal := corev1.SecurityContext{
+		Capabilities:             &corev1.Capabilities{Add: []corev1.Capability{"ALL"}},
+		AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+		Privileged:               &privileged,
+		ReadOnlyRootFilesystem:   &readOnlyFileSystem,
+		RunAsNonRoot:             &runAsNonRoot,
+	}
+
+	terminationGracePeriodSeconds := int64(0)
+
+	serviceDiscoveryDaemonSet := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cr.Namespace,
+			Name:      "submariner-lighthouse-agent",
+			Labels:    labels,
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{MatchLabels: matchLabels},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "submariner-lighthouse-agent",
+							Image:           getImagePath(cr, serviceDiscoveryImage),
+							ImagePullPolicy: "IfNotPresent",
+							SecurityContext: &security_context_all_cap_allow_escal,
+							VolumeMounts: []corev1.VolumeMount{
+								{Name: "host-slash", MountPath: "/host", ReadOnly: true},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "SUBMARINER_NAMESPACE", Value: cr.Spec.Namespace},
+								{Name: "SUBMARINER_CLUSTERID", Value: cr.Spec.ClusterID},
+								{Name: "SUBMARINER_EXCLUDENS", Value: "submariner,kube-system,operators"},
```

Comment:
Shouldn't this also pass the _BROKER_K8S__* env vars?

Reply:
Yes, I have added it.